Kitodo-API/src/main/java/org/kitodo/api/persistentidentifier/PersistentIdentifierGeneratorInterface.java

```diff
@@ -16,10 +16,12 @@
 
     /**
      * Registers the given identifier.
-     * 
+     *
      * @param urn
      *            the identifier to register.
      * @return true, if successful, false otherwise.
+     * @throws UnsupportedOperationException
```

Comment:
It is interface, it doesn't need to have in JavaDoc UnsupportedOperationException. How can you be sure that there is not other implementation which implements this method?

Reply:
There is at least one (the current) implementation that actually throws this exception. Documenting it means:
* to the users of the interface: that they must be aware of the fact that implementations may throw this (unchecked) exception (it is not an error) and handle the situation gracefully
* to implementers of this interface: that in the described case, they should throw *this* exception, and not some other.

See [Iterator.remove()](https://docs.oracle.com/javase/8/docs/api/java/util/Iterator.html#remove--) for a corresponding example in the Java API.